be/dtm0_log.h

```diff
@@ -0,0 +1,235 @@
+ *    the participant sends the persistent notice to rest of the participants
+ *    and originator stating that distributed transaction(dtx) is persistent on
+ *    its store. Upon receiving persistent notice each of the participants and
+ *    originator will update the state of distributed transaction(dtx) for
+ *    remote participant as M0_DTPS_PERSISTENT in DTM0 log record.
+ *
+ *    There is one special case where a process got a persistent notice from
+ *    another process but the corresponding request and reply were not
+ *    received/processed yet. In this case, the DTM service should add a special
+ *    log record: a log record with empty payload (NULL). The users of DTM0 log
+ *    should be ready to encounter such a log entry, and treat it differently
+ *    whenever it is required.
+ */
+
+enum m0_be_dtm0_log_credit_op {
+	M0_DTML_CREATE,		/**< m0_be_dtm0_log_create() */
+	M0_DTML_SENT,		/**< m0_be_dtm0_log_update() */
+	M0_DTML_EXECUTED,	/**< m0_be_dtm0_log_update() */
+	M0_DTML_PERSISTENT,	/**< m0_be_dtm0_log_update() */
+	M0_DTML_PRUNE,		/**< m0_be_dtm0_log_prune()  */
+	M0_DTML_REDO            /**< m0_be_dtm0_log_update() */
+};
+
+struct m0_dtm0_log_rec {
+	struct m0_dtm0_dtx     dlr_dtx;
+	struct m0_dtm0_tx_desc dlr_txd;
+	struct m0_be_list_link dlr_link;    /**< link into m0_be_dtm0_log::tlist */
+	uint64_t               dlr_magic;
+	struct m0_tlink        dlr_tlink;
+	struct m0_buf          dlr_payload;
+};
+
+struct m0_be_dtm0_log {
+	/** Indicates if the structure is a persistent or volatile */
+	bool                     dl_is_persistent;
+	/** DTM0 lock, volatile-field */
+	struct m0_mutex          dl_lock;
+	struct m0_be_seg	*dl_seg;
+	/** DTM0 clock source */
+	struct m0_dtm0_clk_src	*dl_cs;
+	/** Persistent list, used if dl_is_persistent */
+	struct m0_be_list	*dl_list;
+	/** Volatile list, used if !dl_is_persistent */
+	struct m0_tl		*dl_tlist;
+};
+
+/**
+ * DTM log interface typical usecases:
+ * * Typical use cases:
+ * ** Initialisation/Finalisation for volatile fields of the log
+ * - m0_be_dtm0_log_init()
+ * ** Preparation phase
+ * - m0_be_dtm0_log_credit()
+ * ** Normal operation phase
+ * - m0_be_dtm0_log_update()
+ * - m0_be_dtm0_log_prune()
+ * - m0_be_dtm0_log_find()
+ */
+
+M0_INTERNAL int m0_be_dtm0_log_init(struct m0_be_dtm0_log **log,
+                                    struct m0_dtm0_clk_src *cs,
+                                    bool                    isvstore);
+M0_INTERNAL void m0_be_dtm0_log_fini(struct m0_be_dtm0_log **log,
+                                     bool                    isvstore);
+M0_INTERNAL void m0_be_dtm0_log_credit(enum m0_be_dtm0_log_credit_op op,
+				       struct m0_dtm0_tx_desc	    *txd,
+				       struct m0_buf		    *payload,
+                                       struct m0_be_seg             *seg,
+				       struct m0_dtm0_log_rec	    *rec,
+                                       struct m0_be_tx_credit       *accum);
```

Comment:
The parameters of this function declaration are not aligned like others.

Reply:
Done.